editor/plugins/sprite_frames_editor_plugin.cpp

```diff
@@ -949,13 +949,16 @@ void SpriteFramesEditor::_animation_name_edited() {
 	String name = new_name;
 	int counter = 0;
 	while (frames->has_animation(name)) {
+		if (name == String(edited_anim)) {
+			edited->set_text(0, name); // The name didn't change, just updated the column text to name.
+			return;
+		}
```

Comment:
Must be coded out of the while.

Reply:
This is for the case:
"animation_4" => "animation"
And "animation", "animation_1" ... "animation_3" already exists

https://github.com/godotengine/godot/assets/30386067/74c7c5f3-ef3e-4f0a-a772-b39e36a7b562

Reply:
Indeed. Then this would need to be placed "after" the while loop considering the following cases:

1. There is anim, anim1, anim3, anim4
2. Rename anim4 -> anim
3. After while loop it should be anim2, means renamed

At this time, checks during the while loop are redundant.

Reply:
For another case:
1. There is anim, anim_1, anim_2, anim_4
2. Rename anim_2 -> anim 
3. Without this check, it might be renamed to anim_3
4. The result: anim, anim_1, anim_3, anim_4


https://github.com/godotengine/godot/assets/30386067/5ce988db-9547-4a93-92c5-823af5b620d5

When renaming, the old name already exists, so it will be skipped.

Reply:
Oh, you are right. However, this suffix method is also used for state_machine, so the same fix is needed there as well.